src/functions/add_book.rb

```diff
@@ -0,0 +1,13 @@
+require_relative '../book'
+
+def add_book
+  print 'Publisher: '
+  publisher = gets.chomp
+  print 'Cover State (good, bad): '
+  cover_state = gets.chomp
+  print 'Publish Date (YYYY-MM-DD): '
+  publish_date = gets.chomp
+  book = Book.new(publisher, cover_state, publish_date)
+  books << book
+  puts 'Book added successfully'
+end
```

Comment:
With this implementation, there is no way to add a label to this book object and the "List all label" option is always going to return nothing. I kindly suggest that you ask the user to input a label and then call the add label method to set the label.

```bash
  print 'Title: '
  title = gets.chomp
  print 'Color: '
  color = gets.chomp
  label = Label.new(title, color)
  labels << label
  book.add_label(label)
```